landoapi/uplift.py

```diff
@@ -252,45 +299,44 @@ def create_uplift_bug_update_payload(
     return payload
 
 
-UPDATE_RETRIES = 3
-
-
 def update_bugs_for_uplift(
-    bmo_url: str,
-    bmo_api_key: str,
-    changeset_titles: list[str],
     repo_name: str,
-    milestone: int,
+    hgrepo: HgRepo,
+    changeset_titles: list[str],
 ):
     """Update Bugzilla bugs for uplift."""
-    headers = {"X-Bugzilla-API-Key": bmo_api_key}
-    bugs = [str(bug) for title in changeset_titles for bug in parse_bugs(title)]
+    # Parse bug numbers from commits in the stack.
+    bug_ids = [str(bug) for title in changeset_titles for bug in parse_bugs(title)]
+
+    if not bug_ids:
+        raise ValueError("No bugs found in uplift landing.")
+
     params = {
-        "ids": ",".join(bugs),
+        "id": ",".join(bug_ids),
     }
-    bug_endpoint = f"{bmo_url}/rest/bug"
-    resp_get = requests.get(bug_endpoint, headers=headers, params=params)
-    resp_get.raise_for_status()
 
-    bugs = resp_get.json()["bugs"]
+    # Get information about the parsed bugs.
+    bugs = bmo.get_bug(params).json()["bugs"]
+
+    # Get the major release number from `config/milestone.txt`.
+    milestone = get_milestone_major_version(hgrepo)
 
     for bug in bugs:
         payload = create_uplift_bug_update_payload(bug, repo_name, milestone)
```

Comment:
> Good call. Note that this won't actually do what you're suggesting since `map` returns an iterator, so we will still be making requests here before all the payloads are constructed. We would also need to create a lambda wrapper to fill in the other arguments to `create_uplift_bug_update_payload`. I've switched it to a list comprehension which will give us the desired behaviour.

Ah, I mentioned this earlier in my edit in parentheses -- by evaluate, I meant to wrap it in `list` for example, or do a list comprehension. Either way, would be good to fail before we start making calls